hawtio-system/src/main/java/io/hawt/web/FileUploadFilter.java

```diff
@@ -0,0 +1,171 @@
+package io.hawt.web;
+
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.StrictMath.toIntExact;
+
+public class FileUploadFilter {
+
+    private static final transient Logger LOG = LoggerFactory.getLogger(FileUploadFilter.class);
+    public static final String FILE_UPLOAD_PROPNAME = "hawtio.config.fileUpload";
+    private List<MagicNumberFileFilter> filters;
+
+    private void addFilters(MagicNumberFileFilter config) {
```

Comment:
Why is it private when the `getFilters` method is public?  Should users use `getFilters` to add a custom filter?

Reply:
The idea wasn't meant to allow users to create custom filters programmatically rather allowing a singleton way of accessing the filters, this is primarily for `FileUploadFilter` itself to make these filters available for `accept` method, so as long as users have their filters configured via system property then it should just work.